src/wilson/dom.cljs

```diff
@@ -174,9 +174,15 @@
   (update elem 1 merge-attrs attrs))
 
 (defn with-class
-  "Adds a class to an element."
-  [cls elem]
-  (with-attrs {:class cls} elem))
+  "Adds a class to a component."
+  [cls component]
+  (if (vector? component)
```

Comment:
You don't care if this is a vector, you care if it's a vector where the first element is a keyword.

The easiest way to do this is with core.match. Then you can also easily handle the case for `[:a]`, `[:a {:class "whatever"}]`, `[:a {:id "whatever"}]`, `[:a [:div] [:div]]`...